MechJeb2/MechJebModuleStagingController.cs

```diff
@@ -94,23 +94,27 @@ public override void OnUpdate()
             if (HasStayingChutes(StageManager.CurrentStage - 1, vessel))
                 return;
 
-            //only fire decouplers to drop deactivated engines or tanks or fairings
-            bool firesDecoupler = InverseStageFiresDecoupler(StageManager.CurrentStage - 1, vessel);
-            if (firesDecoupler && !(InverseStageDecouplesDeactivatedEngineOrTank(StageManager.CurrentStage - 1, vessel) || HasFairing(StageManager.CurrentStage - 1, vessel)))
-                return;
-
-            //only decouple fairings if the dynamic pressure and altitude conditions are respected
-            if ((core.vesselState.dynamicPressure > fairingMaxDynamicPressure || core.vesselState.altitudeASL < fairingMinAltitude) &&
-                HasFairing(StageManager.CurrentStage - 1, vessel))
-                return;
+            //Always drop deactivated engines or tanks
+            if (!InverseStageDecouplesDeactivatedEngineOrTank(StageManager.CurrentStage - 1, vessel))
+            {
+                //only decouple fairings if the dynamic pressure and altitude conditions are respected
+                if ((core.vesselState.dynamicPressure > fairingMaxDynamicPressure || core.vesselState.altitudeASL < fairingMinAltitude) &&
+                    HasFairing(StageManager.CurrentStage - 1, vessel))
+                    return;
+
+                //only release launch clamps if we're at nearly full thrust
+                if (vesselState.thrustCurrent / vesselState.thrustAvailable < 1/1.2 &&

```

Comment:
Having 2nd thoughts:
1. More real-world rockets than I thought have SLT<1 without SRBs (e.g. all max-payload Atlas V configurations with any SRBs)
2. There's no easy way to make the (fundamentally zero-lookahead, greedy) autostaging work with spooling up multiple sets of engines.
3. SRBs don't seem to have spool up time in RO anyway (?)
4. Checking % available thrust will actually do the right thing in every situation I can think of *without* an engine failure
5. Engine failures are probably too complicated to handle easily and correctly here (even checking SLT is no guarantee due to reduced performance/thrust failures)

So I think % thrust is the right check. I'll make it >1/1.1 (91%) to *guarantee* SLT>1 for simple 1st stages. That doesn't have a noticeable impact due to pre-stage delay but it's good for piece of mind I guess :-)